src/components/BB-BUS/Authority.js

```diff
@@ -40,71 +43,7 @@ export const Authority = ({ data, accordion, onPress, bottomDivider, openWebScre
       />
       {!!accordion[id] && (
         <WrapperWithOrientation>
-          <OperatingCompanyInfo
-            address={(() => {
-              if (!addresses || !addresses.length) return null;
-
-              let address = addresses[0];
-
-              if (address.address) address = address.address;
-
-              const { street, houseNumber, zipcode, city } = address;
-
-              return {
-                street: (!!street || !!houseNumber) && `${address.street} ${address.houseNumber}`,
-                zip: !!zipcode && zipcode,
-                city: !!city && city
-              };
-            })()}
-            contact={(() => {
-              if (!communications || !communications.length) return null;
-
-              let phone = filter(communications, (communication) => {
-                // fix for multi nested result form Directus API
-                if (communication.communication) communication = communication.communication;
-
-                return communication.type.key === 'TELEFON';
-              })[0];
-              // fix for multi nested result form Directus API
-              if (phone && phone.communication) phone = phone.communication;
-
-              let fax = filter(communications, (communication) => {
-                // fix for multi nested result form Directus API
-                if (communication.communication) communication = communication.communication;
-
-                return communication.type.key === 'FAX';
-              })[0];
-              // fix for multi nested result form Directus API
-              if (fax && fax.communication) fax = fax.communication;
-
-              let email = filter(communications, (communication) => {
-                // fix for multi nested result form Directus API
-                if (communication.communication) communication = communication.communication;
-
-                return communication.type.key === 'EMAIL';
-              })[0];
-              // fix for multi nested result form Directus API
-              if (email && email.communication) email = email.communication;
-
-              let www = filter(communications, (communication) => {
-                // fix for multi nested result form Directus API
-                if (communication.communication) communication = communication.communication;
-
-                return communication.type.key === 'WWW';
-              })[0];
-
-              // fix for multi nested result form Directus API
-              if (www && www.communication) www = www.communication;
-
-              return {
-                phone: !!phone && phone.value,
-                fax: !!fax && fax.value,
-                email: !!email && email.value,
-                www: !!www && www.value
-              };
-            })()}
-            openWebScreen={openWebScreen}
-          />
+          <InfoCard address={address} contact={contact} openWebScreen={openWebScreen} />
```

Comment:
wow, this was exactly the same except of the need for `getAddress` and `getContact`?

Reply:
i am not exactly sure what you mean, but there was a lot of duplicated code between `Authority.js` and `Persons.js`

Reply:
for bus bb these two methods are necessary to provide proper objects because of the weird nesting of directus graphql responses, right? without them it wouldn't work. so i was happy, that everything else works the same.